extensions/amp-apester-media/0.1/monetization/companion/video.js

```diff
@@ -0,0 +1,173 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Services} from '../../../../../src/services';
+const ALLOWED_AD_PROVIDER = 'sr';
+import {createElementWithAttributes} from '../../../../../src/dom';
+import {dict} from '../../../../../src/utils/object';
+import {getValueForExpr} from '../../../../../src/json';
+
+/**
+ * @param {!JsonObject} media
+ * @param {AmpElement} apesterElement
+ * @param {!JsonObject} consentObj
+ */
+export function handleCompanionVideo(media, apesterElement, consentObj) {
+  const companionCampaignOptions = getValueForExpr(
+    /**@type {!JsonObject}*/ (media),
+    'campaignData.companionCampaignOptions'
+  );
+  const videoSettings = getValueForExpr(
+    /**@type {!JsonObject}*/ (media),
+    'campaignData.companionOptions.video'
+  );
+  const position = getCompanionPosition(
+    /**@type {!JsonObject}*/ (videoSettings)
+  );
+
+  if (
+    !companionCampaignOptions ||
+    !videoSettings ||
+    !videoSettings['enabled'] ||
+    videoSettings['provider'] !== ALLOWED_AD_PROVIDER ||
+    !position ||
+    position === 'floating'
+  ) {
+    return;
+  }
+  const macros = getSrMacros(
+    media,
+    companionCampaignOptions['companionCampaignId'],
+    apesterElement,
+    consentObj
+  );
+  constructCompanionSrElement(
+    videoSettings['videoTag'],
+    position,
+    /** @type {!JsonObject} */ (macros),
+    apesterElement
+  );
+}
+
+/**
+ * @param {!JsonObject} video
+ * @return {?string}
+ */
+function getCompanionPosition(video) {
+  if (video) {
+    if (video['companion']['enabled']) {
+      return 'above';
+    }
+    if (video['companion_below']['enabled']) {
+      return 'below';
+    }
+    if (video['floating']['enabled']) {
+      return 'floating';
+    }
+  }
+}
+
+/**
+ * @param {string} videoTag
+ * @param {string} position
+ * @param {!JsonObject} macros
+ * @param {AmpElement} apesterElement
+ */
+function constructCompanionSrElement(
+  videoTag,
+  position,
+  macros,
+  apesterElement
+) {
+  const size = getCompanionVideoAdSize(apesterElement);
+  const ampBladeAd = createElementWithAttributes(
+    /** @type {!Document} */ (apesterElement.ownerDocument),
+    'amp-ad',
+    dict({
+      'width': size.width,
+      'height': '0',
+      'type': 'blade',
+      'layout': 'responsive',
+      'data-blade_player_type': 'bladex',
+      'servingDomain': 'ssr.streamrail.net',
+      'data-blade_macros': JSON.stringify(macros),
+      'data-blade_player_id': videoTag,
+      'data-blade_api_key': '5857d2ee263dc90002000001',
+    })
+  );
+
+  ampBladeAd.classList.add('amp-apester-companion');
+
+  const relativeElement =
+    position === 'below' ? apesterElement.nextSibling : apesterElement;
+  apesterElement.parentNode.insertBefore(ampBladeAd, relativeElement);
+
+  apesterElement.getResources().attemptChangeSize(ampBladeAd, size.height);
```

Comment:
This doesn't seem quite right, since you are using `layout="responsive"`, the width will be 100% of the parent (which can scale up or down), but the height will be a static size.

Not sure the best way forward here.

Reply:
OK, thanks I'll change it to be layout fixed that anyway was the first intent.